examples/cc3200/xively_demo/main.c

```diff
@@ -0,0 +1,1127 @@
+                xi_schedule_timed_task( in_context_handle, send_temprature, 5, 1, NULL );
+
+            if ( XI_INVALID_TIMED_TASK_HANDLE == gSendTempratureHandle )
+            {
+                UART_PRINT( "send_temprature_task couldn't be registered\r\n" );
+            }
+
+            /* re-subscribe to topics led topics */
+            xi_subscribe( in_context_handle, gGreenLedTopicName, XI_MQTT_QOS_AT_MOST_ONCE,
+                          onLedTopic, ( void* )&gGreenLed );
+            xi_subscribe( in_context_handle, gOrangeLedTopicName,
+                          XI_MQTT_QOS_AT_MOST_ONCE, onLedTopic, ( void* )&gOrangeLed );
+            xi_subscribe( in_context_handle, gRedLedTopicName, XI_MQTT_QOS_AT_MOST_ONCE,
+                          onLedTopic, ( void* )&gRedLed );
+
+            /* re-enable the button interrupts */
+            Button_IF_EnableInterrupt( SW2 | SW3 );
+
+            break;
+        case XI_CONNECTION_STATE_CLOSED:
+            UART_PRINT( "connection closed - reason %d!\n", state );
+
+            /* cancel timed task - we don't want to send messages while the library not
+             * connected */
+            xi_cancel_timed_task( gSendTempratureHandle );
+            gSendTempratureHandle = XI_INVALID_TIMED_TASK_HANDLE;
+
+            /* disable button interrupts so the messages for button push/release events
+             * won't be sent */
+            Button_IF_DisableInterrupt( SW2 | SW3 );
+
+            /* check if the close was due to the user's request */
+            if ( XI_STATE_OK == state )
+            {
+                xi_events_stop();
+            }
+            else
+            {
+                waitForWlanEvent();
+
+                xi_connect_to( in_context_handle, conn_data->host, conn_data->port,
+                               conn_data->username, conn_data->password,
+                               conn_data->connection_timeout,
+                               conn_data->keepalive_timeout, conn_data->session_type,
+                               &on_connected );
+            }
+            return;
+        default:
+            UART_PRINT( "invalid parameter %d\n", conn_data->connection_state );
+            return;
+    }
+}
+
+/**
+ * @brief Driver for the xively library logic
+ *
+ * It initialises the library, creates a context and starts the internal event dispatcher
+ * and event processing of xively library.
+ *
+ * This function is blocking for the sakness of this demo application however it is
+ * possible to run xively's event processing with limit of iterations.
+ */
+void ConnectToXively()
+{
+    /* Register Push Button Handlers */
+    Button_IF_Init( pushButtonInterruptHandlerSW2, pushButtonInterruptHandlerSW3 );
+
+    // Make the buttons to react on both press and release events
+    MAP_GPIOIntTypeSet( GPIOA1_BASE, GPIO_PIN_5, GPIO_BOTH_EDGES );
+    MAP_GPIOIntTypeSet( GPIOA2_BASE, GPIO_PIN_6, GPIO_BOTH_EDGES );
+
+    // Disable the interrupt for now

```

Comment:
/* */  Not that it's required here, because the platform's toolchain supports both after all, but I hate mixing and matching the comment type in the same function.